iotilecore/iotile/core/scripts/virtualdev_script.py

```diff
@@ -134,8 +140,10 @@ def main(argv=None, loop=SharedLoop):
             raise
 
         started = True
-
         print("Starting to serve virtual IOTile device")
+        if device_f:
```

Comment:
I think for safety you might also want to check the device existing. Technically we should be raising if the instantiation fails, but it's easy to check regardless:
```suggestion
        if device_f and device is not None:
```